plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs

```diff
@@ -124,9 +125,14 @@ beta
 beta = \case
     -- See Note [Multi-beta]
     -- This maybe isn't the best annotation for this term, but it will do.
-    (extractBindings -> Just (bs, t)) -> Let (termAnn t) NonRec bs (beta t)
+    (extractBindings -> Just (bs, t)) ->
```

Comment:
Here's a possible alternative that might be nicer. It makes the transformation part simpler and I think makes the pattern of "do the local transform and then recurse" clearer.

```
localTransform = \case
  (extractBindings -> ...
  TyInst ...
  t -> t
beta t = (localTransform t) & termSubterms %~ beta
```

Reply:
Good suggestion. Updated.